apps/server/src/modules/task/service/task.service.ts

```diff
@@ -73,6 +73,15 @@ export class TaskService {
 			this.authorizationService.checkPermission(user, course, PermissionContextBuilder.write([]));
 			taskParams.course = course;
 
+			if (params.lessonId) {
+				const lesson = await this.lessonRepo.findById(params.lessonId);
+				if (!taskParams.course || lesson.course.id !== taskParams.course.id) {
```

Comment:
this `!taskParams.course` still stiches for me. I dont feel well to throw "Lesson does not belong to course" if there is actually no course present.
which in itself is a question if it can happen at all, as in line 74 this is only written to if we have a courseId present. so, can that happen in the first place?

edit: there is no also no specific test for that case, or do I just miss that?

Reply:
was moved back (outside) the course param condition